### PR TITLE
Unbreak the ops.relation changes, use the new Prometheus libraries

### PR DESCRIPTION
**Pull request by rbarry82**

### CODE REVIEW

**Review by balbirthomas — APPROVED** [review]

File: lib/charms/cassandra_k8s/v0/cassandra.py [hunk 1]
```diff
@@ -10,41 +10,22 @@
...
-acceptable version of Cassandra. This version string can be in any format that
-is compatible with the Python [Semantic Version
-module](https://pypi.org/project/semantic-version/). For example, assuming your
-charm consumes a database over a rlation named "monitoring", you may
-instantiate `CassandraConsumer` as follows
+Charms that would like to use a Cassandra database must use the `CassandraConsumer`
+object from the charm library. Using the `CassandraConsumer` object requires
+instantiating it, typically in the constructor of your charm. The `CassandraConsumer`
+constructor requires the name of the relation over which a database will be used.
+This relation must use the `cassandra` interface.
```
Comment (sed-i) [inline lib/charms/cassandra_k8s/v0/cassandra.py:17]: Need to update metadata.yaml? It still has `cql`. And the "cassandra" relation's interface is "prometheus_scrape". I am quite confused.

**Review by sed-i — APPROVED** [review]

File: lib/charms/cassandra_k8s/v0/cassandra.py [hunk 2]
```diff
@@ -217,16 +192,17 @@ def restore(self, snapshot):
...
-class CassandraConsumer(ConsumerBase):
+class CassandraConsumer(Object):
     on = CassandraConsumerEvents()
 
-    def __init__(self, charm, name, consumes, multi=False):
-        super().__init__(charm, name, consumes, multi)
+    def __init__(self, charm, name):
+        super().__init__(charm, name)
         self.charm = charm
+        self.name = name
```
Comment (dstathis) [inline lib/charms/cassandra_k8s/v0/cassandra.py]: why? it is never used

File: lib/charms/cassandra_k8s/v0/cassandra.py [hunk 3]
```diff
@@ -339,12 +315,13 @@ class CassandraProviderEvents(ObjectEvents):
...
-class CassandraProvider(ProviderBase):
+class CassandraProvider(Object):
     on = CassandraProviderEvents()
 
-    def __init__(self, charm, name, service, version=None):
-        super().__init__(charm, name, service, version)
+    def __init__(self, charm, name):
+        super().__init__(charm, name)
         self.charm = charm
+        self.name = name
```
Comment (dstathis) [inline lib/charms/cassandra_k8s/v0/cassandra.py:323]: why?

File: lib/charms/cassandra_k8s/v0/cassandra.py (hunk 1 above)
Comment (dstathis) [reply lib/charms/cassandra_k8s/v0/cassandra.py:17]: It should say cql here. I'll change in a later PR

**Review by dstathis — CHANGES_REQUESTED** [review]

**Review by dstathis — APPROVED** [review]

**Pull request merged**